### PR TITLE
Fix usage of foo.encode() and foo.decode() for Python 3

### PR DESCRIPTION
**Pull request by rodrigc**

### CODE REVIEW

File: master/buildbot/changes/hgpoller.py [hunk 1]
```diff
@@ -111,7 +111,9 @@ def _getRevDetails(self, rev):
         @d.addCallback
         def process(output):
             # all file names are on one line
-            date, author, files, comments = output.decode(self.encoding, "replace").split(
+            if isinstance(output, bytes):
```
Comment (tardyp) [inline master/buildbot/changes/hgpoller.py]: output = ascii2unicode(output)

File: master/buildbot/data/types.py [hunk 2]
```diff
@@ -129,7 +129,10 @@ class String(Instance):
     ramlType = "string"
 
     def valueFromString(self, arg):
-        return arg.decode('utf-8')
+        val = arg
```
Comment (tardyp) [inline master/buildbot/data/types.py]: ditto

File: master/buildbot/data/types.py [hunk 3]
```diff
@@ -163,7 +166,9 @@ def __init__(self, len=None, **kwargs):
         self.len = len
 
     def valueFromString(self, arg):
-        val = arg.decode('utf-8')
+        val = arg
```
Comment (tardyp) [inline master/buildbot/data/types.py]: ditto

File: master/buildbot/steps/worker.py [hunk 4]
```diff
@@ -154,10 +154,10 @@ def commandComplete(self, cmd):
     # TODO: BuildStep subclasses don't have a describe()....
     def getResultSummary(self):
         src = self.src
-        if isinstance(src, str):
+        if isinstance(src, bytes):
```
Comment (tardyp) [inline master/buildbot/steps/worker.py]: we can safely replace ascii decode by utf8 decode here

File: master/buildbot/test/fake/fakedb.py [hunk 5]
```diff
@@ -107,7 +107,8 @@ def __init__(self, **kwargs):
         # which stores either utf-8 encoded string, or gzip-compressed
         # utf-8 encoded string.
         for col in self.binary_columns:
-            self.values[col] = self.values[col].encode("utf-8")
+            if isinstance(self.values[col], text_type):
```
Comment (tardyp) [inline master/buildbot/test/fake/fakedb.py]: idem

File: master/buildbot/data/types.py (hunk 3 above)
Comment (rodrigc) [inline master/buildbot/data/types.py]: Replacing with ascii2unicode doesn't work here.  Running the tests
I got decoding errors.  I will come up with another solution.

**Comment by rodrigc** [comment]
OK, I came up with a a patch that works.

File: master/buildbot/util/__init__.py [hunk 6]
```diff
@@ -220,16 +220,20 @@ def none_or_str(x):
     return x
 
 
-def ascii2unicode(x):
+def unicode2bytes(x, encoding='utf-8', errors='strict'):
```
Comment (tardyp) [inline master/buildbot/util/__init__.py]: great!

**Pull request merged**